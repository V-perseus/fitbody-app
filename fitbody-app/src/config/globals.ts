import { Dimensions, TextProps } from 'react-native'
import styles from './styles'
import string from './strings'
import { FB_CURRENT_ENV, FB_ENVIRONMENTS } from '../../env'

export const globals = {
  // Styles & Colors
  styles: {
    ...styles,
  },
  // Fonts
  fonts: {
    ...styles.fonts,
  },
  // Header Styles
  header: {
    icons: {
      marginRight: 15,
      marginLeft: 15,
    },
    headerTitleStyle: <TextProps>{
      fontSize: 16,
      fontFamily: styles.fonts.primary.semiboldStyle.fontFamily,
      textAlign: 'center',
    },
    headerTruncatedBackTitle: true,
  },
  // Standard String
  strings: {
    ...string,
  },
  // Background Images
  bgImage: {
    imageStyle: { resizeMode: 'cover' },
    style: {
      width: Dimensions.get('window').width,
      height: Dimensions.get('window').height,
    },
  },
  // Window dimensions
  window: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  },
  // Button click targeting
  touchableOpacity: {
    activeOpacity: 0.8,
    hitSlop: { top: 5, right: 5, left: 5, bottom: 5 },
  },
  fbEnvironments: {
    ...FB_ENVIRONMENTS,
  },
  currentEnv: FB_CURRENT_ENV,
  get apiBase(): string {
    return this.fbEnvironments[this.currentEnv]
  },
}

export default globals
