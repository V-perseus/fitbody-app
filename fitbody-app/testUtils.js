import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
