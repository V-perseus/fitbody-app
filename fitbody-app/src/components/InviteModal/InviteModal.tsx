import React from 'react'
import { View } from 'react-native'

import JoinAmbassadorModal from '../JoinAmbassadorModal'
import ReferralProgramModal from '../ReferralProgramModal'

import globals from '../../config/globals'
import { useAppSelector } from '../../store/hooks'
import { MenuScreenUseNavigationProp } from '../../config/routes/routeTypes'

interface IInviteModalProps {
  navigation: MenuScreenUseNavigationProp
}
const InviteModal: React.FC<IInviteModalProps> = ({ navigation }) => {
  const {
    user: { ambassador_link: ambassadorLink, is_ambassador: isAmbassador },
  } = useAppSelector((state) => {
    return state.data
  })

  return (
    <View style={globals.styles.container}>
      {isAmbassador && ambassadorLink ? (
        <ReferralProgramModal navigation={navigation} ambassadorCode={ambassadorLink} />
      ) : (
        <JoinAmbassadorModal navigation={navigation} />
      )}
    </View>
  )
}

export default InviteModal
