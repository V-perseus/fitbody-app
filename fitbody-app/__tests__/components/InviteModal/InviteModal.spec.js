import 'react-native'
import React from 'react'
import { expect, it } from '@jest/globals'

import { render } from '../../../testUtils'
import InviteModal from '../../../src/components/InviteModal'

describe('InviteModal', () => {
  it('renders snapshot correctly', () => {
    const tree = render(<InviteModal />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders JoinAmbassadorModal content correctly', () => {
    const { getByText } = render(<InviteModal />)

    const headerText = getByText(/Loving the Fit Body app/i)
    const joinBtn = getByText('JOIN NOW')

    expect(headerText).not.toBeNull()
    expect(joinBtn).not.toBeNull()
  })

  it('renders ReferralProgramModal content correctly', () => {
    const { getByText } = render(<InviteModal />, {
      preloadedState: { data: { user: { is_ambassador: true, ambassador_link: 'TEST LINK' } } },
    })

    const headerText = getByText(/Share this link to invite a friend/i)
    const shareBtn = getByText('SHARE YOUR LINK')
    const link = getByText('TEST LINK')

    expect(headerText).not.toBeNull()
    expect(shareBtn).not.toBeNull()
    expect(link).not.toBeNull()
  })
})
