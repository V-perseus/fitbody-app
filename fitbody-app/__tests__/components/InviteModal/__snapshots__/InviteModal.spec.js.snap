// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteModal renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      forwardedRef={[Function]}
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 8,
            "marginHorizontal": 16,
            "paddingHorizontal": 25,
            "paddingVertical": 31,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={16}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "opacity": 1,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "position": "absolute",
                "right": 8,
                "top": 8,
              },
            ]
          }
        >
          <close.svg
            color="#333333"
          />
        </View>
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 18,
              "marginBottom": 23,
              "textAlign": "center",
            }
          }
        >
          Loving the Fit Body app and 
 want to invite a friend?
        </Text>
        <View
          style={
            Object {
              "borderBottomColor": "#fa8dd5",
              "borderBottomWidth": 2,
              "borderTopColor": "#fa8dd5",
              "borderTopWidth": 2,
              "marginBottom": 24,
              "paddingVertical": 11,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fa8dd5",
                "fontFamily": "BebasNeue",
                "fontSize": 45,
                "letterSpacing": 0,
                "textAlign": "center",
              }
            }
          >
            GIVE 25%. GET 25%.
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        >
          For every friend who signs up through your referral link, you get 25% commission on their first payment, and your friend gets 25% off their first payment!*

Bonus! For every 20 successful referrals, get an additional $20! 
 
 Join now to get your own share link and start earning today!
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 20,
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "opacity": 1,
                },
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "height": 56,
                  "justifyContent": "center",
                  "width": 114,
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fa8dd5",
                  "borderRadius": 8,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 48,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                  undefined,
                ]
              }
            >
              JOIN NOW
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontSize": 12,
              "marginTop": 16,
              "textAlign": "center",
            }
          }
        >
          *Friend must be a new member. Commission will be approved once friend makes their first payment.
        </Text>
      </View>
    </View>
  </View>
</View>
`;
